api/src/controllers/zot4planimport.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ @module Zot4PlanImportRoute
+*/
+
+import { z } from 'zod';
+import { publicProcedure, router } from '../helpers/trpc';
+import { SavedRoadmap, SavedPlannerQuarterData, QuarterName } from '@peterportal/types';
+
+type Zot4PlanSchedule = {
+  years: string[][][];
+  selectedPrograms: {
+    value: number;
+    label: string;
+    is_major: boolean;
+  }[][];
+  addedCourses: [];
+  courses: [];
+  apExam: {
+    id: number;
+    name: string;
+    score: number;
+    courses: string[];
+    GE: [];
+    units: number;
+  }[];
+};
+
+/**
+ * Get a JSON schedule from Zot4Plan by name; return an empty {} if it does not exist
+ */
+const getFromZot4Plan = async (scheduleName: string) => {
+  let res = {};
+  await fetch('https://api.zot4plan.com/api/loadSchedule/' + scheduleName, {
+    method: 'PUT',
+  })
+    .then((response) => {
+      if (!response.ok) {
+        return {};
+      }
```

Comment:
If we have an issue, I think it'd be better to return an error rather than an empty object. That way, we can tell the use instead of having something silently fail